geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/ListGatewayCommand.java

```diff
@@ -53,9 +53,21 @@ public ResultModel listGateway(
           help = CliStrings.LIST_GATEWAY__MEMBER__HELP) String[] onMember,
       @CliOption(key = {CliStrings.GROUP, CliStrings.GROUPS},
           optionContext = ConverterHint.MEMBERGROUP,
-          help = CliStrings.LIST_GATEWAY__GROUP__HELP) String[] onGroup)
+          help = CliStrings.LIST_GATEWAY__GROUP__HELP) String[] onGroup,
+      @CliOption(key = {CliStrings.LIST_GATEWAY__RECEIVERS_ONLY},
+          specifiedDefaultValue = "false", unspecifiedDefaultValue = "true",
+          help = CliStrings.LIST_GATEWAY__RECEIVERS_ONLY) boolean getSenders,
+      @CliOption(key = {CliStrings.LIST_GATEWAY__SENDERS_ONLY},
+          specifiedDefaultValue = "false", unspecifiedDefaultValue = "true",
+          help = CliStrings.LIST_GATEWAY__SENDERS_ONLY__HELP) boolean getReceivers)
+
       throws Exception {
 
+    if (!getSenders && !getReceivers) {
```

Comment:
Looks like this is going to return both senders and receivers list even when user has explicitly set the values to false. Either we should throw error saying one needs to be true or return empty set.

Also, for the following case, due to specified default value "false"; it will list the receivers, wherein the expectation may be only senders.

> list gateways --senders

Reply:
+1 to @agingade's comments above.

Reply:
Notice that option `--senders` set the value for `getReceivers` variable, and `--receivers` set the value for `getSenders`. 

That is:
* When `list gateways` then `getReceivers=true` & `getSenders=true` 
* When `list gateways --senders` then `getReceivers=false` & `getSenders=true`
* When `list gateways --receivers` then `getReceivers=true` & `getSenders=false`
* When `list gateways --senders --receivers` then `getReceivers=false` & `getSenders=false`, so I set them to true, to get both senders & receivers, because for me its more intuitive. I would expect to get both senders and receivers if I introduce both parameters.

Thanks to your comment @agingade I realize there is a missing assert in the `testListGatewaySenderOnly` test, because Im not checking if the output has or not the gateway receiver section, as it is done in `testListGatewayReceiverOnly`.

Reply:
I did not notice the inverted use of attributes...
For "sender" argument it is setting "getReceivers"
Can we change it to "noSender" that way we know its associated with the sender....

For "list gateways --senders --receivers"
Agree with you; it should list out both.

For "list gateways --senders=false --receivers=false"
Its not expected, but one could execute it; in that case, we just need to ignore the command...And list empty set. Or a message saying either one of them is required.

Reply:
@alb3rtobr please let me know if you need any help.

Reply:
@agingade Thanks, I have not been able to spend more time on this until today.